pkg/cloud/services/networking/service.go

```diff
@@ -78,3 +83,23 @@ func NewTestService(projectID string, client NetworkClient, logger logr.Logger)
 		logger:    logger,
 	}
 }
+
+// replaceAllAttributesTags replaces all tags on a neworking resource.
+// the value of resourceType must match one of the allowed constants: trunkResource or portResource.
+func (s *Service) replaceAllAttributesTags(eventObject runtime.Object, resourceType string, resourceID string, tags []string) error {
+	if resourceType != trunkResource && resourceType != portResource {
+		record.Warnf(eventObject, "FailedReplaceAllAttributesTags", "Invalid resourceType argument in function call")
+		return fmt.Errorf("invalid argument: resourceType, %s, does not match allowed arguments: %s or %s", resourceType, trunkResource, portResource)
+	}
+
```

Comment:
is it even worth checking it?

OpenStack would return an error on wrong types and we have it hardcoded.
Also if we add it, wouldn't it make sense to implement it in the `networkClient.ReplaceAllAttributesTags`?

But also ok for me as is.